src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py

```diff
@@ -640,6 +657,73 @@ def test_containerapp_cors_policy(self, resource_group):
+        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+
+        env_name = self.create_random_name(prefix='containerapp-env', length=24)
+        logs_workspace_name = self.create_random_name(prefix='containerapp-env', length=24)
+
+        logs_workspace_id = self.cmd(f'monitor log-analytics workspace create -g {resource_group} -n {env_name} -l eastus'.format(resource_group, logs_workspace_name)).get_output_in_json()["customerId"]
+        logs_workspace_key = self.cmd(f'monitor log-analytics workspace get-shared-keys -g {resource_group} -n {env_name}'.format(resource_group, logs_workspace_name)).get_output_in_json()["primarySharedKey"]
+
+        self.cmd(f'containerapp env create -g {resource_group} -n {env_name} --logs-workspace-id {logs_workspace_id} --logs-workspace-key {logs_workspace_key}')
+
+        containerapp_env = self.cmd(f'containerapp env show -g {resource_group} -n {env_name}').get_output_in_json()
+
+        self.cmd(f'az containerapp env workload-profile add -g {resource_group} -n {env_name} -w wp-ingress --min-nodes 2 --max-nodes 5 --workload-profile-type D4'.format(env_name, resource_group))
+
+        self.cmd(f'containerapp env premium-ingress show -g {resource_group} -n {env_name}', checks=[
+            JMESPathCheck('message', 'No premium ingress configuration found for this environment, using default values.'),
+        ])
+
+        self.cmd(f'containerapp env premium-ingress add -g {resource_group} -n {env_name} -w wp-ingress --min-replicas 3 --max-replicas 5', checks=[
+            JMESPathCheck('workloadProfileName', 'wp-ingress'),
+            JMESPathCheck('scale.minReplicas', 3),
+            JMESPathCheck('scale.maxReplicas', 5),
+            JMESPathCheck('terminationGracePeriodSeconds', None),
+            JMESPathCheck('requestIdleTimeout', None),
+            JMESPathCheck('headerCountLimit', None),
+        ])
+        
+        self.cmd(f'containerapp env premium-ingress show -g {resource_group} -n {env_name}', checks=[
+            JMESPathCheck('workloadProfileName', 'wp-ingress'),
+            JMESPathCheck('scale.minReplicas', 3),
+            JMESPathCheck('scale.maxReplicas', 5),
+            JMESPathCheck('terminationGracePeriodSeconds', None),
+            JMESPathCheck('requestIdleTimeout', None),
+            JMESPathCheck('headerCountLimit', None),
+        ])
+        
+        self.cmd(f'containerapp env premium-ingress update -g {resource_group} -n {env_name} --min-replicas 4 --max-replicas 20 --termination-grace-period 45 --request-idle-timeout 180 --header-count-limit 40', checks=[
+            JMESPathCheck('workloadProfileName', 'wp-ingress'),
+            JMESPathCheck('scale.minReplicas', 4),
+            JMESPathCheck('scale.maxReplicas', 20),
+            JMESPathCheck('terminationGracePeriodSeconds', 45),
+            JMESPathCheck('requestIdleTimeout', 180),
+            JMESPathCheck('headerCountLimit', 40),
+        ])
```

Comment:
what is `containerapp env premium-ingress update` don't update some of the existing params, will they be updated?

Seems `containerapp env premium-ingress update` and `containerapp env premium-ingress add` are the same and duplicate?

Reply:
Add has a few required parameters for the initial setup, and it removes any settings you don't specify. Update has no required parameters, it only modifies the ones you specify and keeps the existing settings for the others.